src/providers/ipa/ipa_subdomains_id.c

```diff
@@ -506,7 +506,13 @@ struct tevent_req *ipa_get_subdom_acct_send(TALLOC_CTX *memctx,
             break;
         default:
             ret = EINVAL;
-            DEBUG(SSSDBG_OP_FAILURE, "Invalid sub-domain request type.\n");
+            if (state->entry_type > BE_REQ_BY_CERT) {
```

Comment:
Using `BE_REQ_BY_CERT` explicit here might be a bit risky if new requests are added. Maybe something like `BE_REQ_LAST` or similar can be added to `src/providers/data_provider_req.h` with a comment that it should always point to the REQ with the highest number?

In general it looks like an enum would be better suited for `BE_REQ_*` which e.g. would avoid the gap in the list, but this should not be part of this patch set.

Reply:
Done.